pkg/utils/termination/suite_test.go

```diff
@@ -0,0 +1,113 @@
+	"testing"
+
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+
+	"sigs.k8s.io/karpenter/pkg/cloudprovider/fake"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "knative.dev/pkg/logging/testing"
+
+	"sigs.k8s.io/karpenter/pkg/apis"
+	"sigs.k8s.io/karpenter/pkg/operator/scheme"
+	. "sigs.k8s.io/karpenter/pkg/test/expectations"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+	"sigs.k8s.io/karpenter/pkg/test"
+)
+
+var (
+	ctx           context.Context
+	env           *test.Environment
+	cloudProvider *fake.CloudProvider
+)
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "NodeClaimUtils")
+}
+
+var _ = BeforeSuite(func() {
+	env = test.NewEnvironment(scheme.Scheme, test.WithCRDs(apis.CRDs...))
+	cloudProvider = fake.NewCloudProvider()
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = AfterEach(func() {
+	cloudProvider.Reset()
+	ExpectCleanedUp(ctx, env.Client)
+})
+
+var _ = Describe("NodeClaimUtils", func() {
+	var nodeClaim *v1beta1.NodeClaim
+	BeforeEach(func() {
+		nodeClaim = test.NodeClaim()
+		cloudProvider.CreatedNodeClaims[nodeClaim.Status.ProviderID] = nodeClaim
+	})
+	It("should update the status condition on NodeClaim as Terminating", func() {
+		nodeClaim.StatusConditions().SetFalse(v1beta1.ConditionTypeTerminating, "NotTerminating", "NotTerminating")
+		ExpectApplied(ctx, env.Client, nodeClaim)
+		instanceTerminated, err := EnsureTerminated(ctx, env.Client, nodeClaim, cloudProvider)
+
+		Expect(instanceTerminated).To(BeFalse())
+		Expect(err).NotTo(HaveOccurred())
+		Expect(nodeClaim.StatusConditions().Get(v1beta1.ConditionTypeTerminating).IsTrue()).To(BeTrue())
+	})
+	It("should check if the cloudProvider instance is terminated", func() {
+		nodeClaim.StatusConditions().SetTrue(v1beta1.ConditionTypeTerminating)
+		ExpectApplied(ctx, env.Client, nodeClaim)
+		instanceTerminated, err := EnsureTerminated(ctx, env.Client, nodeClaim, cloudProvider)
+
+		Expect(instanceTerminated).To(BeFalse())
+		Expect(err).NotTo(HaveOccurred())
+	})
+	It("should return true when the cloudProvider instance is terminated", func() {
+		nodeClaim.StatusConditions().SetFalse(v1beta1.ConditionTypeTerminating, "NotTerminating", "NotTerminating")
+		ExpectApplied(ctx, env.Client, nodeClaim)
+		// This will call cloudProvider.Delete()
+		instanceTerminated, err := EnsureTerminated(ctx, env.Client, nodeClaim, cloudProvider)
+		Expect(instanceTerminated).To(BeFalse())
+		Expect(err).NotTo(HaveOccurred())
+
+		//This will call cloudProvider.Get(). Instance is terminated at this point
+		instanceTerminated, err = EnsureTerminated(ctx, env.Client, nodeClaim, cloudProvider)
+
+		Expect(instanceTerminated).To(BeTrue())
+		Expect(err).NotTo(HaveOccurred())
+	})
+	It("should return true if nodeClaim is not found", func() {
+		nodeClaim.StatusConditions().SetFalse(v1beta1.ConditionTypeTerminating, "NotTerminating", "NotTerminating")
```

Comment:
Same here

Reply:
How different is this test from the previous test in this suite file?

Reply:
This is different because we get nodeClaimNotFoundError when we make cloudProvider Delete call which is different from the other test. The other test first makes the delete call and then makes a get call and get returns nodeClaimNotFoundError.